cypress/integration/regression/main-table-regression.spec.ts

```diff
@@ -0,0 +1,226 @@
+/* eslint-disable no-undef */
+import { CommonSelectors } from '../../../src/components/data-testid/common-selectors.cy'
+import { datasetPage } from '../../page-objects/app/datasets-page'
+import { mainTablePage } from '../../page-objects/app/main-table-page'
+import { variantDrawerPage } from '../../page-objects/app/variant-drawer-page'
+import { Timeouts } from '../../page-objects/lib/timeouts.cy'
+
+describe('Regression test of the main table | step 1', () => {
+  const link = '/ws?ds=PGP3140_wgs_panel_hl'
+  const homozygous = '⏚BGM_Homozygous_Rec'
+
+  it('should open main table for PGP3140_wgs_panel_hl dataset', () => {
+    datasetPage.visit()
+    datasetPage.leftPanel.leftPanelHeader.checkLabelText('Datasets')
+    datasetPage.leftPanel.datasetsListElem
+      .getButtonByText('PGP3140_wgs_panel_hl')
+      .click()
+    datasetPage.leftPanel.leftPanelHeader.checkLabelText('Datasets')
+    cy.wait(500)
+    datasetPage.datasetInfo.openInViewer.click()
+    datasetPage.datasetInfo.mainTable.click()
+    cy.url().should('include', '/ws?ds=PGP3140_wgs_panel_hl')
+  })
+
+  it('should apply preset on a dataset | step 2', () => {
+    mainTablePage.visit(link)
+    cy.intercept('POST', '/app/tab_report').as('dsUpload')
+    cy.wait('@dsUpload', { timeout: 20000 })
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2592')
+    mainTablePage.mainTable.selectPreset.click()
+    cy.intercept('POST', '/app/tab_report').as('loadPreset')
+    mainTablePage.mainTable.preset.getButtonByText(homozygous).click()
+    cy.wait('@loadPreset', { timeout: 20000 })
+    //number of variants changes
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2')
+  })
+
+  it('should apply gene to the preset | step 3', () => {
+    mainTablePage.visit(link)
+    cy.intercept('POST', '/app/tab_report').as('dsUpload')
+    cy.wait('@dsUpload', { timeout: 20000 })
+    mainTablePage.mainTable.selectPreset.click()
+    cy.intercept('POST', '/app/tab_report').as('loadPreset')
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2592')
+    mainTablePage.mainTable.preset.getButtonByText(homozygous).click()
+    cy.wait('@loadPreset', { timeout: Timeouts.TwentySecondsTimeout })
+    //number of variants changes
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2')
+    mainTablePage.mainTable.addGene.click()
+    mainTablePage.mainTable.searchFilter.type('CHSY1')
+    mainTablePage.mainTable.geneCheckbox.check()
+    mainTablePage.mainTable.applyButton.click()
```

Comment:
You can create a method in `mainTable` for these 4 instructions, because I see you're reusing them. It could be something like that: `updateFilter(options: string[]) => {}`